### PR TITLE
Fixes in CCP for #1228

### PR DESCRIPTION
**Pull request by alan-baker**
Addresses #1228 

* Forces traversal of phis if the def has changed to varying
* Mark a phi as varying if all incoming values are varying
* added a test to catch the bug

I will wait for a review from @dnovillo before submitting.

### CODE REVIEW

File: test/opt/ccp_test.cpp [hunk 1]
```diff
@@ -581,6 +581,66 @@ TEST_F(CCPTest, SkipSpecConstantInstrucitons) {
   auto res = SinglePassRunToBinary<opt::CCPPass>(spv_asm, true);
   EXPECT_EQ(std::get<1>(res), opt::Pass::Status::SuccessWithoutChange);
 }
+
+TEST_F(CCPTest, UpdateSubsequentPhisToVarying) {
```
Comment (dnovillo) [inline test/opt/ccp_test.cpp]: This test does not seem to have any CHECKs?

Comment (alan-baker) [reply test/opt/ccp_test.cpp]: Correct its testing the pass succeeds with no change.

**Comment by dnovillo** [comment]
On Mon, Jan 29, 2018 at 2:47 PM, alan-baker <notifications@github.com>
wrote:

> *@alan-baker* commented on this pull request.
> ------------------------------
>
> In test/opt/ccp_test.cpp
> <https://github.com/KhronosGroup/SPIRV-Tools/pull/1233#discussion_r164543474>
> :
>
> > @@ -581,6 +581,66 @@ TEST_F(CCPTest, SkipSpecConstantInstrucitons) {
>    auto res = SinglePassRunToBinary<opt::CCPPass>(spv_asm, true);
>    EXPECT_EQ(std::get<1>(res), opt::Pass::Status::SuccessWithoutChange);
>  }
> +
> +TEST_F(CCPTest, UpdateSubsequentPhisToVarying) {
>
> Correct its testing the pass succeeds with no change.
>

Thanks.  I clearly need more vacation ;)

LGTM.

**Comment by alan-baker** [comment]
Rebased and committed in 1b46f7ecad32f99e0b28702505a192d7bf680bd7

Fixes #1228

**Pull request closed without merging**